File: CareConnect/CareConnect.Models/Requests/WorkshopUpdateRequest.cs
using System;
using System.Text.Json.Serialization;

namespace CareConnect.Models.Requests
{
    public class WorkshopUpdateRequest
    {
        public string? Name { get; set; } = null!;

        public string? Description { get; set; } = null!;

        public string? WorkshopType { get; set; }

        public DateTime? StartDate { get; set; }

        public DateTime? EndDate { get; set; }

        public decimal? Price { get; set; }

        public decimal? MemberPrice { get; set; }

        public int? MaxParticipants { get; set; }

        public int? Participants { get; set; }

        public string? Notes { get; set; }

        [JsonIgnore]
        public DateTime ModifiedDate { get; set; } = DateTime.Now;
    }
}

File: UI/careconnect_mobile/lib/models/requests/workshop_update_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'workshop_update_request.g.dart'; // Don't open or edit this

@JsonSerializable()
class WorkshopUpdateRequest {
  String? name;
  String? description;
  String? workshopType;
  DateTime? startDate;
  DateTime? endDate;
  double? price;
  double? memberPrice;
  int? maxParticipants;
  int? participants;
  String? notes;

  WorkshopUpdateRequest({
    this.name,
    this.description,
    this.workshopType,
    this.startDate,
    this.endDate,
    this.price,
    this.memberPrice,
    this.maxParticipants,
    this.participants,
    this.notes,
  });

  factory WorkshopUpdateRequest.fromJson(Map<String, dynamic> json) =>
      _$WorkshopUpdateRequestFromJson(json);

  Map<String, dynamic> toJson() => _$WorkshopUpdateRequestToJson(this);
}
